### ISSUE TITLE
After upgrading from 21.8 to 22.8, the CPU usage doubles.

### ISSUE THREAD

**Issue by beijishiqidu** [ClickHouse/ClickHouse#55593]
After I upgraded Clickhouse from 21.8.8.29 to 22.8.11.15, I tested the performance with the previous model and found that the CPU usage went straight up to 99%. After I rolled back to 21.8.8.29, the CPU usage dropped to 50%.

I have 400 tables. Each table contains more than 300 fields. All of them are of the String type and can be null. The tested model does not change. The cluster configuration did not change, but the CPU usage doubled.

CPU：24core
Mem：172G；

The cluster configuration information is as follows:

config.xml
```xml
<?xml version="1.0"?>
<yandex>
    <logger>
        <level>trace</level>
        <log>/srv/clickhouse/log/clickhouse-server.log</log>
        <errorlog>/srv/clickhouse/log/clickhouse-server.err.log</errorlog>
        <size>1000M</size>
        <count>10</count>
    </logger>
    <max_table_size_to_drop>0</max_table_size_to_drop>
    <max_server_memory_usage>0</max_server_memory_usage>
    <max_partition_size_to_drop>0</max_partition_size_to_drop>
    <listen_host>172.18.0.151</listen_host>
    <max_connections>8192</max_connections>
    <keep_alive_timeout>3</keep_alive_timeout>
    <max_concurrent_queries>2000</max_concurrent_queries>
    <max_open_files>1048576</max_open_files>
    <uncompressed_cache_size>88589934592</uncompressed_cache_size>
    <mark_cache_size>55368709120</mark_cache_size>
    <path>/opt/clickhouse/data/</path>
    <tmp_path>/opt/clickhouse/tmp/</tmp_path>
    <user_files_path>/opt/clickhouse/user_files/</user_files_path>
    <users_config>/opt/clickhouse/config/users.xml</users_config>
    <default_profile>default</default_profile>
    <mlock_executable>true</mlock_executable>
    <remote_servers incl="clickhouse_remote_servers" ></remote_servers>
    <include_from>/opt/clickhouse/config/metrika.xml</include_from>
    <zookeeper incl="zookeeper-servers" optional="true" />
    <macros incl="macros" optional="true" />
    <builtin_dictionaries_reload_interval>3600</builtin_dictionaries_reload_interval>
    <max_session_timeout>3600</max_session_timeout>
    <default_session_timeout>60</default_session_timeout>
    <query_log>1</query_log>
    <trace_log>0</trace_log>
    <query_thread_log>0</query_thread_log>
    <query_log>
        <database>system</database>
        <table>query_log</table>
        <partition_by>toYYYYMM(event_date)</partition_by>
        <ttl>event_date + INTERVAL 1 MONTH DELETE</ttl>
        <flush_interval_milliseconds>7500</flush_interval_milliseconds>
    </query_log>
    <trace_log>
        <database>system</database>
        <table>trace_log</table>
        <partition_by>toYYYYMM(event_date)</partition_by>
        <flush_interval_milliseconds>7500</flush_interval_milliseconds>
    </trace_log>
    <query_thread_log>
        <database>system</database>
        <table>query_thread_log</table>
        <partition_by>toYYYYMM(event_date)</partition_by>
        <flush_interval_milliseconds>7500</flush_interval_milliseconds>
    </query_thread_log>
    <dictionaries_config>*_dictionary.xml</dictionaries_config>
    <compression incl="clickhouse_compression"></compression>
    <distributed_ddl>
        <path>/opt/clickhouse/task_queue/ddl</path>
    </distributed_ddl>
    <prometheus>
        <endpoint>/metrics</endpoint>
        <port>26732</port>
        <metrics>true</metrics>
        <events>true</events>
        <asynchronous_metrics>true</asynchronous_metrics>
        <status_info>true</status_info>
<ssl>true</ssl>
    </prometheus>
    <graphite_rollup_example>
        <pattern>
            <regexp>click_cost</regexp>
            <function>any</function>
            <retention>
                <age>0</age>
                <precision>3600</precision>
            </retention>
            <retention>
                <age>86400</age>
                <precision>60</precision>
            </retention>
        </pattern>
        <default>
            <function>max</function>
            <retention>
                <age>0</age>
                <precision>60</precision>
            </retention>
            <retention>
                <age>3600</age>
                <precision>300</precision>
            </retention>
            <retention>
                <age>86400</age>
                <precision>3600</precision>
            </retention>
        </default>
    </graphite_rollup_example>
    <format_schema_path>/opt/clickhouse/format_schemas/</format_schema_path>
   <storage_configuration>
      <disks>
         <default>
            <keep_free_space_bytes>53687091200</keep_free_space_bytes>
         </default>
      </disks>
   </storage_configuration>
    <merge_tree>
        <old_parts_lifetime>10</old_parts_lifetime>
        <temporary_directories_lifetime>10</temporary_directories_lifetime>
        <allow_floating_point_partition_key>1</allow_floating_point_partition_key>
    </merge_tree>
        <openSSL>
              <server>
                    <certificateFile>/opt/clickhouse/pem/DataCubeServer.pem</certificateFile>
                    <privateKeyFile>/opt/clickhouse/pem/DataCubeServer.key</privateKeyFile>
                    <verificationMode>none</verificationMode>
                    <loadDefaultCAFile>false</loadDefaultCAFile>
                    <cacheSessions>true</cacheSessions>
                    <disableProtocols>sslv2,sslv3,tls1.0,tls1.1</disableProtocols>
                    <cipherList>ECDHE-RSA-AES256-GCM-SHA384</cipherList>
                    <preferServerCiphers>true</preferServerCiphers>
              </server>
              <client>
                    <caConfig>/opt/clickhouse/pem/DataCubeTrust.pem</caConfig>
                    <loadDefaultCAFile>false</loadDefaultCAFile>
                    <cacheSessions>true</cacheSessions>
                    <disableProtocols>sslv2,sslv3,tls1.0,tls1.1</disableProtocols>
                    <preferServerCiphers>true</preferServerCiphers>
              </client>
        </openSSL>
        <https_port>31943</https_port>
        <tcp_port_secure>31940</tcp_port_secure>
</yandex>

```

users.xml
```xml
<?xml version="1.0" encoding="UTF-8" standalone="no"?>
<yandex>
    <profiles>
        <default>
            <skip_unavailable_shards>1</skip_unavailable_shards>
            <max_memory_usage>0</max_memory_usage>
            <max_memory_usage_for_user>0</max_memory_usage_for_user>
            <use_uncompressed_cache>0</use_uncompressed_cache>
            <load_balancing>random</load_balancing>
            <send_timeout>3600</send_timeout>
            <receive_timeout>3600</receive_timeout>
            <group_by_two_level_threshold>100000</group_by_two_level_threshold>
            <group_by_two_level_threshold_bytes>50000000</group_by_two_level_threshold_bytes>
            <max_query_size>1073741824</max_query_size>
            <max_ast_elements>1000000</max_ast_elements>
            <max_expanded_ast_elements>1000000</max_expanded_ast_elements>
        </default>
    </profiles>
    <users>
        <writer>
            <networks incl="networks" replace="replace">
                <ip>::/0</ip>
            </networks>
            <profile>default</profile>
            <quota>default</quota>
        </writer>
        <inner_sync>
            <networks incl="networks" replace="replace">
                <ip>::/0</ip>
            </networks>
            <profile>default</profile>
            <quota>default</quota>
        </inner_sync>
    </users>
    <quotas>
        <default>
            <interval>
                <duration>3600</duration>
                <queries>0</queries>
                <errors>0</errors>
                <result_rows>0</result_rows>
                <read_rows>0</read_rows>
                <execution_time>0</execution_time>
            </interval>
        </default>
    </quotas>
</yandex>

```

metrika.xml
```xml
<yandex>
  <clickhouse_remote_servers>
    <GDECluster>
      <shard>
        <internal_replication>false</internal_replication>
        <replica>
          <host>clickhouse-0</host>
          <port>26730</port>
          <user>inner_sync</user>
          <password>IuHO.D6l2mDKqxHs</password>
          <secure>1</secure>
        </replica>
        <replica>
          <host>clickhouse-1</host>
          <port>26730</port>
          <user>inner_sync</user>
          <password>IuHO.D6l2mDKqxHs</password>
          <secure>1</secure>
        </replica>
      </shard>
    </GDECluster>
  </clickhouse_remote_servers>
</yandex>

```

I did not use zookeeper or clickhouse-keeper for metadata management. I used a distributed engine, and the two nodes were replicas. Data is directly inserted into the distributed table during database import. The cluster automatically synchronizes data. I remember clickhouse had this ability.

**Comment by den-crane** [comment]
22.8.11.15 ???  Why not 22.8.21.38 ?
Or 23.8.3.48 ?

**Comment by den-crane** [comment]
22.8 is out of support

**Comment by beijishiqidu** [comment]
@den-crane Hello, my friend, For some reason, we can't use this version yet, but we'll upgrade it as soon as possible, but is this a known issue? This problem is solved in 22.8.21.38 or later versions.

**Comment by den-crane** [comment]
> @den-crane Hello, my friend, For some reason, we can't use this version yet, but we'll upgrade it as soon as possible, but is this a known issue? This problem is solved in 22.8.21.38 or later versions.

There are a lot of bugs and performance issues is in 22.8.11.
No sense to spend anyone time (including your time) to debug issues probably solved a long time ago.
